components/dashboard/src/repositories/detail/prebuilds/PrebuildSettingsForm.tsx

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { BranchMatchingStrategy, Configuration } from "@gitpod/public-api/lib/gitpod/v1/configuration_pb";
+import { FC, FormEvent, useCallback, useState } from "react";
+import { ConfigurationSettingsField } from "../ConfigurationSettingsField";
+import { Heading3, Subheading } from "@podkit/typography/Headings";
+import { InputField } from "../../../components/forms/InputField";
+import { PartialConfiguration, useConfigurationMutation } from "../../../data/configurations/configuration-queries";
+import { useToast } from "../../../components/toasts/Toasts";
+import { SelectInputField } from "../../../components/forms/SelectInputField";
+import { TextInputField } from "../../../components/forms/TextInputField";
+import { WorkspaceClassOptions } from "../shared/WorkspaceClassOptions";
+import { LoadingButton } from "@podkit/buttons/LoadingButton";
+
+type Props = {
+    configuration: Configuration;
+};
+
+export const PrebuildSettingsForm: FC<Props> = ({ configuration }) => {
+    const { toast } = useToast();
+    const updateConfiguration = useConfigurationMutation();
+
+    const [interval, setInterval] = useState<string>(`${configuration.prebuildSettings?.prebuildInterval ?? 20}`);
```

Comment:
Should we have these magic constants like the prebuild interval or the default machine class somewhere as a variable to which we can refer? It feels like we do it on a lot of places